src/omnicore/rpctx.cpp

```diff
@@ -1085,14 +1085,20 @@ Value omni_sendalert(const Array& params, bool fHelp)
 
     // obtain parameters & info
     std::string fromAddress = ParseAddress(params[0]);
-    int32_t alertType = params[1].get_int();
-    uint64_t expiryValue = params[2].get_uint64();
-    uint32_t typeCheck = params[3].get_uint64();
-    uint32_t verCheck = params[4].get_uint64();
-    std::string alertMessage = ParseText(params[5]);
+    int64_t tempAlertType = params[1].get_int64();
+    if (tempAlertType < 1 || 65535 < tempAlertType) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Alert type is out of range");
+    }
+    uint16_t alertType = static_cast<uint16_t>(tempAlertType);
+    int64_t tempExpiryValue = params[2].get_int64();
+    if (tempExpiryValue < 1 || 4294967295LL < tempExpiryValue) {
```

Comment:
With the [current version schema](https://github.com/OmniLayer/omnicore/pull/162/files#diff-da586973b290a2920a045b8ad5ddf4b9R47) we shall never reach 1.0.0.0, hehe. ;)